src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsExecution.java

```diff
@@ -67,8 +79,104 @@ public boolean start() throws Exception {
 
     @Override
     public void stop(@Nonnull Throwable cause) throws Exception {
-        if (body != null)
+        if (body != null) {
             body.cancel(cause);
+        }
+    }
+
+    /**
+     * Finds a template from the given templates list that matches the given label String.
+     *
+     * @param diskPoolRefId   the disk pool ref id that the found template should have
+     * @param nodeLabelString the label that the template has
+     * @param templates       a list of templates
+     * @return the template matching the given label, <code>null</code> otherwise
+     * @throws IOException if the found template doesn't have defined a disk pool ref id
+     *                     if the defined disk pool ref id doesn't match the one given as parameter
+     */
+    @CheckForNull
+    private static Template findTemplate(String diskPoolRefId, String nodeLabelString, List<Template> templates) throws IOException {
+        Template selectedTemplate = null;
+        for (Template template : templates) {
+            String templateLabel = template.getLabel();
+            if (templateLabel != null && nodeLabelString.contains(templateLabel)) {
+                selectedTemplate = template;
+                break;
+            }
+        }
+        if (selectedTemplate != null) {
```

Comment:
You could replace this line with an early return `if (selectedTemplate == null) return null`, but no strong opinion here.